packages/mendel-middleware/index.js

```diff
@@ -29,6 +30,38 @@ function MendelMiddleware(opts) {
             return getPath({ bundle: bundle, hash: tree.hash });
         };
 
+        req.mendel.getExposed = function(variations) {
+            if (!variations) {
+                variations = 'base';
+            }
+
+            if (exposedBundleVariations[variations]) {
+                return exposedBundleVariations[variations];
+            }
+
+            var exposedBundleVariation = Object.keys(trees.bundles).reduce(function(acc, bundle) {
+                var vTree = trees.findTreeForVariations(bundle, variations);
```

Comment:
This is the bottleneck for performance. We can't call this on every bundle on every request. Also, we don't need variations for that.